### ISSUE TITLE
Error `'MapboxDirections/MapboxDirections.h' file not found` after installing MapboxNavigation

### ISSUE THREAD

**Issue by paul019** [mapbox/mapbox-directions-swift#318]
Hi there,

I've installed MapboxNavigation iOS via CocoaPods without experiencing any issues. But after compiling the project in Xcode I got two errors produced by the library. One error is `'MapboxDirections/MapboxDirections.h' file not found`:

<img width="1792" alt="bildschirmfoto 2018-11-05 um 20 40 35" src="https://user-images.githubusercontent.com/39464035/48135795-3499b980-e29e-11e8-8f69-9c0895b80c03.png">

I tried to just comment out the line of code that produced the error and the whole project indeed compiled successfully. As I can evaluate the import that produces the error isn't needed in the file at all. Is this true?

After doing this hacky bugfix, I started writing some lines of Mapbox code and got another error:

**The code:**
```
import Foundation
import UIKit
import MapboxCoreNavigation
import MapboxNavigation
import MapboxDirections
  ...
let navigationService = MapboxNavigationService(route: route, simulating: .always)
let navigationController = NavigationViewController(for: route, navigationService: navigationService)
viewController.addChildViewController(navigationController);
viewController.addSubview(navigationController.view);
```

**The resulting error:**
`Command failed due to signal: Abort trap: 6`

I've no clue what this error means and think that it may have something to do with my hacky bugfix. Is this possible?

**Mapbox Navigation SDK version:** 0.22.0

**Comment by mysport12** [comment]
We are also running into this issue on our project. use_frameworks! causes other pods to fail. We are using MapboxNavigation v26 and Cocoapods v1.5.3.

**Comment by paul019** [comment]
@frederoni When can we expect a solution to this issue?

**Comment by frederoni** [comment]
It's not a long-term solution but I was able to work around this issue by renaming `MapboxDirections.swift` to `MapboxDirections` and using the following Podfile:

<details>
<summary>Podfile</summary>

```ruby
source 'https://github.com/CocoaPods/Specs.git'
platform :ios, '9.0'

target 'my-app' do
  pod 'MapboxCoreNavigation', :path => '../mapbox-navigation-ios'
  pod 'MapboxNavigation', :path => '../mapbox-navigation-ios'
  pod 'MapboxMobileEvents', :modular_headers => true
  pod 'MapboxDirections', :path => '../MapboxDirections.swift', :modular_headers => true
end
```
</details>

along with renaming `MapboxDirections.swift.podspec` to `MapboxDirections.podspec` and edit `MapboxCoreNavigation.podspec` to depend on `MapboxDirections` instead of `MapboxDirections.swift`.

I know there was an issue with `.` in framework names in Xcode 6/7, but the bug seems to have resurfaced, at least with static frameworks.

**Comment by paul019** [comment]
@frederoni Thank you for your answer! I'm very sorry for not answering for such a long time.

I've now tried out your workaround, but unfortunately the `pod repo update && pod install` fails for me, if I use your Podfile. That's the error message that is thrown in the console:

`[!] No podspec found for 'MapboxCoreNavigation' in '../mapbox-navigation-ios'`

So, what's the problem here?

**Comment by 1ec5** [comment]
The Podfile in https://github.com/mapbox/MapboxDirections.swift/issues/318#issuecomment-450863548 assumes you have a clone of the dependencies’ repositories beside the current project directory.

**Comment by 1ec5** [comment]
> static libraries wrapped in a framework is the default in CocoaPods (as of 1.5?), supported on Xcode >= 9 and iOS >= 9. Any idea why MapboxDirections.swift’s umbrella header would be missing unless `use_frameworks!` is specified?

It’s unusual that MapboxDirections-Swift.h tries to import MapboxDirections.h. By contrast, the generated interface for MapboxSpeech, MapboxSpeech.h, lacks this import statement.

```objc
#import <MapboxDirections/MapboxDirections.h>
```

There should be no reason for a library’s generated interface header to import the same library’s umbrella header. I think this may come down to the fact that MapboxDirections.swift’s pod name (`MapboxDirections.swift`) doesn’t match the product module name (`MapboxDirections`) and possibly that it contains “.swift”, which makes it look like a file.

**Comment by aliustaoglu** [comment]
I cannot use frameworks because of other dependencies. And as per https://github.com/mapbox/MapboxDirections.swift/issues/318#issuecomment-450863548 I renamed `MapboxDirections.swift.podspec` to `MapboxDirections.podspec` but still no luck. I get this error:

Redefinition of module 'MapboxDirections'

**Comment by 1ec5** [comment]
> There should be no reason for a library’s generated interface header to import the same library’s umbrella header. I think this may come down to the fact that MapboxDirections.swift’s pod name (MapboxDirections.swift) doesn’t match the product module name (MapboxDirections) and possibly that it contains “.swift”, which makes it look like a file.

As of #400 in v1.0.0-alpha.1, the pod’s name matches the module name, so the umbrella header should no longer be broken. What’s more, this library no longer supports Objective-C, so the umbrella header likely has even less relevance.

Please check again whether the issue persists in v1.0.0-alpha.1.

**Comment by dwieners** [comment]
Hey there,  I have some issues since iOS 13.3.1 with `use_frameworks!`.
[https://forums.developer.apple.com/thread/128435](https://forums.developer.apple.com/thread/128435)
Is there any way to fix that issue? Unfortunately I can't load `MapboxDirections/MapboxDirections.h`by adding `:modular_headers => true` to my Podfile

**Comment by 1ec5** [comment]
> Hey there,  I have some issues since iOS 13.3.1 with `use_frameworks!`.
[https://forums.developer.apple.com/thread/128435](https://forums.developer.apple.com/thread/128435)

That’s mapbox/mapbox-gl-native-ios#157, unrelated to the problem that was tracked in this issue.

**Comment by 1ec5** [comment]
This library no longer supports Objective-C code directly. See #381 for background and a workaround if your application is still written in Objective-C.

**Comment by 1ec5** [comment]
My bad, I thought the original issue was about importing the umbrella header manually, as one would in Objective-C. Still, I think the root cause mentioned in https://github.com/mapbox/mapbox-directions-swift/issues/318#issuecomment-450863548 is resolved as of v1.0.0-alpha.1, since we’ve removed “.swift” from the pod’s name.
